### ISSUE TITLE
False positive for terraform.azure.security.keyvault.keyvault-ensure-secret-expires.keyvault-ensure-secret-expires when setting from a function or resource

### ISSUE THREAD

**Issue by EsgAtWorkleap** [returntocorp/sgrep-rules#3564]
**Describe the bug**
We are setting an `expiration_date` on a `azurerm_key_vault_secret` using a `time_rotating` resource, which triggers the rule.
Using a function also triggers the rule. Basically anything but a date string will trigger.

**To Reproduce**
```hcl
// using time_rotating
resource "time_rotating" "secret_expiration_date" {
  rotation_days = 365
}

resource "azurerm_key_vault_secret" "secret" {
  key_vault_id    = "..."
  name            = "..."
  value           = "..."
  content_type    = "..."
  expiration_date = time_rotating.secret.rotation_rfc3339
}
```
OR

```hcl
// using a function
resource "azurerm_key_vault_secret" "secret" {
  key_vault_id    = "..."
  name            = "..."
  value           = "..."
  content_type    = "..."
  expiration_date = timeadd("1982-12-31T00:00:00Z", 1h)
}
```

**Expected behavior**
The rule should not trigger if expiration_date is set via a resource or function

**Priority**
How important is this to you?
- [X] P0: blocking me from making progress
- [ ] P1: this will block me in the near future
- [ ] P2: annoying but not blocking me

**Additional Context**

**Comment by LewisArdern** [comment]
this was fixed i believe so closing
